com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/CCWindow/CCWindow_cpp.cgt

```diff
@@ -0,0 +1,253 @@
+/* Additional includes go here */
+
+#include "splpy.h"
+#include "splpy_funcop.h"
+
+using namespace streamsx::topology;
+
+<%SPL::CodeGen::implementationPrologue($model);%>
+
+@include "../pyspltuple.cgt"
+<%
+ # Select the Python wrapper function
+ my $pyoutstyle = splpy_tuplestyle($model->getOutputPortAt(0));
+ 
+ # Temporary, until more input types are supported. At the moment, only pass by reference types 
+ # are supported.
+ my $in_pywrapfunc =  $pystyle . '_in__object_out';
+ my $out_pywrapfunc=  'pickle_in__' . $pyoutstyle . '_out';
+%>
+
+#define SPLPY_TUPLE_MAP(f, v, r, occ) \
+    streamsx::topology::Splpy::pyTupleMap(f, v, r)
+
+// Constructor
+MY_OPERATOR::MY_OPERATOR() :
+   funcop_(NULL),
+   pyInNames_(NULL),
+   occ_(-1),
+   window_(*this, 0,
+   CountWindowPolicy(<%=$model->getParameterByName("evictConfig")->getValueAt(0)->getSPLExpression()%>),
+   CountWindowPolicy(<%=$model->getParameterByName("triggerConfig")->getValueAt(0)->getSPLExpression()%>))
+{
+    window_.registerOnWindowTriggerHandler(this);
+    window_.registerAfterTupleEvictionHandler(this);
+
+    const char * in_wrapfn = "<%=$in_pywrapfunc%>";
+    const char * out_wrapfn = "<%=$out_pywrapfunc%>";
+    const char * functions_module = "streamsx.topology.functions";
+    const char * identity_function = "identity";
+
+<%
+# If occ parameter is positive then pass-by-ref is possible
+# Generate code to allow pass by ref but only use when
+# not connected to a PE output port.
+
+ my $oc = $model->getParameterByName("outputConnections");
+
+ if ($oc) {
+    my $occ = $oc->getValueAt(0)->getSPLExpression();
+    if ($occ > 0) {
+        my $pybyrefwrapfunc = 'pickle_in__object_out';
+%>
+
+#undef SPLPY_TUPLE_MAP
+#define SPLPY_TUPLE_MAP(f, v, r, occ) \
+    streamsx::topology::Splpy::pyTupleMapByRef(f, v, r, occ)
+
+    if (!this->getOutputPortAt(0).isConnectedToAPEOutputPort()) {
+       // pass by reference
+       out_wrapfn = "<%=$pybyrefwrapfunc%>";
+       occ_ = <%=$occ%>;
+    }
+<%
+    } 
+ }
+%>
+
+    funcop_ = new SplpyFuncOp(this, out_wrapfn);
+    spl_in_object_out = new SplpyFuncOp(this, in_wrapfn, functions_module, identity_function);

```

Comment:
This is possibly overkill. The wrapping capability was designed to call a user function, but in this case there is no need to call a user function (`identity`). The real requirement is to convert the raw Python object into the logical Python object of which there will be three cases:
         1. Python bytes to depicked object
         1. Python str to dict as JSON
         1. No need for conversion (Schema is String, Binary, SPL Schema (Python dict))`

With the wrapped user function case (e.g map), there is always a need to call a wrapping function which does optional conversion, calls the user function and handles the return. That's not true in this case.

Since this PR started, there is now an example of code having to call a Python function repeatably, in `MemoryViewCleanup` :
```
static PyObject * releaser = SplpyGeneral::loadFunctionGIL("streamsx.spl.runtime", "_splpy_release_memoryviews");
```

and calling it:

```
PyObject * ret = SplpyGeneral::pyCallObject(releaser(), args);
```



While it is implementation detail, it does end up adding a new `SplpyFuncOp` constructor that is not needed.